File: homework/quang_tu/lesson_14/Grab/task_2.cpp
#include <bits/stdc++.h>
using namespace std;

int solution (vector<int>& A) {
    int n = A.size(),
        i = 0;

    A.push_back(0);

    while (i < n + 1) {
        if (A[i] == i) {
            i++;
        } else {
            if (A[i] > n || A[A[i]] == A[i])
                return 0;

            int index = A[i];
            A[i] = A[A[i]];
            A[index] = index;
        }
    }

    return 1;
}

void solve() {
    int n;
    cin >> n;
    vector<int> A (n);

    for (int i = 0; i < n; ++i)
        cin >> A[i];

    int res = solution(A);

    cout << res << endl;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;

    for (int i = 1; i <= t; ++i) {
        cout << "Case #" << i << ": ";
        solve();
    }

    return 0;
}

File: homework/Truong_Duong/Lesson_13/Task2.js
function solution(S) {
  let res = 0;

  while (S[0] === '0') {
    S = S.slice(1);
  }

  while (S !== '0') {
    const size = S.length;
    if (S[size - 1] === '0') {
      S = S.slice(0, size - 1);
    } else {
      S = S.slice(0, size - 1) + '0';
    }
    res += 1;
  }

  return res;
}

console.log(solution('011100') === 7);
console.log(solution('111') === 5);
console.log(solution('1111010101111') === 22);
